ui/src/data_explorer/components/DatabaseDropdown.js

```diff
@@ -0,0 +1,77 @@
+import React, {PropTypes} from 'react'
+
+import {showDatabases, showRetentionPolicies} from 'shared/apis/metaQuery'
+import showDatabasesParser from 'shared/parsing/showDatabases'
+import showRetentionPoliciesParser from 'shared/parsing/showRetentionPolicies'
+import Dropdown from 'shared/components/Dropdown'
+
+const {func, shape, string} = PropTypes
+
+const DatabaseDropdown = React.createClass({
+  propTypes: {
+    query: shape({}).isRequired,
+    onChooseNamespace: func.isRequired,
+  },
+
+  contextTypes: {
+    source: shape({
+      links: shape({
+        proxy: string.isRequired,
+      }).isRequired,
+    }).isRequired,
+  },
+
+  getInitialState() {
+    return {
+      namespaces: [],
+    }
+  },
+
+  componentDidMount() {
```

Comment:
the current pattern is to use ES7 async/await syntax for `componentDidMount` and its internals, rather than promise syntax, ex. under `dashboards/components/DatabasesDropdown.js`. but i'm ok with this, as there is an outstanding issue to refactor all promise syntax to async/await, and this will be included in that.

either way, we do need to `catch` and handle errors on any AJAX requests, passing to the `errorThrown` action creator.